File: GeneratedEntityReaderClasses/Components/DamageNearbyEntitiesComponent.cs
namespace GeneratedNoitaClasses;
public class DamageNearbyEntitiesComponent : NoitaComponentBase,  iNoitaType<DamageNearbyEntitiesComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat radius { get; set; } = new();
	public NoitaFloat damage_min { get; set; } = new();
	public NoitaFloat damage_max { get; set; } = new();
	public NoitaFloat target_vec_max_len { get; set; } = new();
	public NoitaFloat knockback_multiplier { get; set; } = new();
	public NoitaInt time_between_damaging { get; set; } = new();
	public NoitaUInt damage_type { get; set; } = new();
	public NoitaString damage_description { get; set; } = new();
	public NoitaString target_tag { get; set; } = new();
	public NoitaUInt ragdoll_fx { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		radius.Read(s);
		damage_min.Read(s);
		damage_max.Read(s);
		target_vec_max_len.Read(s);
		knockback_multiplier.Read(s);
		time_between_damaging.Read(s);
		damage_type.Read(s);
		damage_description.Read(s);
		target_tag.Read(s);
		ragdoll_fx.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		radius.Write(s);
		damage_min.Write(s);
		damage_max.Write(s);
		target_vec_max_len.Write(s);
		knockback_multiplier.Write(s);
		time_between_damaging.Write(s);
		damage_type.Write(s);
		damage_description.Write(s);
		target_tag.Write(s);
		ragdoll_fx.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/DamageNearbyEntitiesComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaUInt

from NoitaStream import NoitaStream
@dataclass
class DamageNearbyEntitiesComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    radius:NoitaFloat = field(default_factory=NoitaFloat)
    damage_min:NoitaFloat = field(default_factory=NoitaFloat)
    damage_max:NoitaFloat = field(default_factory=NoitaFloat)
    target_vec_max_len:NoitaFloat = field(default_factory=NoitaFloat)
    knockback_multiplier:NoitaFloat = field(default_factory=NoitaFloat)
    time_between_damaging:NoitaInt = field(default_factory=NoitaInt)
    damage_type:NoitaUInt = field(default_factory=NoitaUInt)
    damage_description:NoitaString = field(default_factory=NoitaString)
    target_tag:NoitaString = field(default_factory=NoitaString)
    ragdoll_fx:NoitaUInt = field(default_factory=NoitaUInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.radius.Read(s)
        self.damage_min.Read(s)
        self.damage_max.Read(s)
        self.target_vec_max_len.Read(s)
        self.knockback_multiplier.Read(s)
        self.time_between_damaging.Read(s)
        self.damage_type.Read(s)
        self.damage_description.Read(s)
        self.target_tag.Read(s)
        self.ragdoll_fx.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.radius.Write(s)
        self.damage_min.Write(s)
        self.damage_max.Write(s)
        self.target_vec_max_len.Write(s)
        self.knockback_multiplier.Write(s)
        self.time_between_damaging.Write(s)
        self.damage_type.Write(s)
        self.damage_description.Write(s)
        self.target_tag.Write(s)
        self.ragdoll_fx.Write(s)

